airbyte-json-validation/src/main/java/io/airbyte/validation/json/JsonSchemaValidator.java

```diff
@@ -58,11 +61,68 @@ public JsonSchemaValidator() {
    * @param baseUri The base URI for schema resolution
    */
   @VisibleForTesting
-  protected JsonSchemaValidator(URI baseUri) {
+  protected JsonSchemaValidator(final URI baseUri) {
     this.jsonSchemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
     this.baseUri = baseUri;
   }
 
+  /**
+   * Create and cache a schema validator for a particular schema. This validator is used when
+   * {@link #testInitializedSchema(String, JsonNode)} and
+   * {@link #ensureInitializedSchema(String, JsonNode)} is called.
+   */
+  public void initializeSchemaValidator(final String schemaName, final JsonNode schemaJson) {
+    schemaToValidators.put(schemaName, getSchemaValidator(schemaJson));
+  }
+
+  /**
+   * Returns true if the object adheres to the given schema and false otherwise.
+   */
+  public boolean testInitializedSchema(final String schemaName, final JsonNode objectJson) {
+    final var schema = schemaToValidators.get(schemaName);
+    Preconditions.checkNotNull(schema, schemaName + " needs to be initialised before calling this method");
```

Comment:
@colesnodgrass instead of making this auto-cache, I'm going to fail loudly for now.

This entire class needs to be rewritten for better methods + auto cache. However, want to do that separately to isolate it from performance work.